python/testing/__init__.py

```diff
@@ -422,6 +422,14 @@ def start_app(cleanup=True):
         QGISAPP = QgsApplication(argvb, myGuiFlag)
 
         QGISAPP.initQgis()
+
+        # Make sure geopackages are not written-to, during tests
+        # See https://github.com/qgis/QGIS/issues/25830
+        # NOTE: this needs to happen _after_ QgsApplication::initQgis()
```

Comment:
it doesn't reset them -- it just correctly initializes the storage so that QgsSettings can be safely used. I think this comment can be dropped

Reply:
If it is unsafe to call QgsSettings before ::initQgis maybe the constructor should throw an exception ? I think it's important for the comment to stay, maybe I'll change the wording, but it's useful information (I spent a long time to find out)